### PR TITLE
disables modal close on backdrop click

### PR DESCRIPTION
**Pull request by tillyw**
## Associated issues
fixes https://github.com/cityofaustin/atd-data-tech/issues/17095

## Testing
**URL to test:** 

https://deploy-preview-1388--atd-moped-main.netlify.app/

**Steps to test:**
- go to a project map page and click 'new component'
- click outside of the new component dialog, nothing should happen
- close out of the component by completing it or clicking the 'x'

---
#### Ship list
- [x] Code reviewed 
- [x] Product manager approved
- [x] Product manager checked [DB view dependencies](https://atd-dts.gitbook.io/moped-documentation/product-management/updating-the-arcgis-online-components-database-view)
~- [ ] Product manager added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable~

### CODE REVIEW

File: moped-editor/src/views/projects/projectView/ProjectComponents/CreateComponentModal.js [hunk 1]
```diff
@@ -76,7 +76,9 @@ const CreateComponentModal = ({
     }
   };
 
-  const onClose = () => {
+  const onClose = (event, reason) => {
+    if (reason && reason === "backdropClick")
```
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectComponents/CreateComponentModal.js:80]: cool to see how this is implemented, and I like that this retains the ability to press `ESC` to exit the modal! I think that, between the esc and the close button (which can be tabbed into focus or clicked), there are plenty of ways to exit the form with intention.

**Review by mddilley — COMMENTED** [review]
Looks and works great! Could you add the same behavior to `EditAttributesModal`?

File: moped-editor/src/views/projects/projectView/ProjectComponents/CreateComponentModal.js (hunk 1 above)
Comment (frankhereford) [reply moped-editor/src/views/projects/projectView/ProjectComponents/CreateComponentModal.js:80]: +1 on the escape -- it was the first thing that I tried when I was experimenting with it. Works great.

**Review by frankhereford — APPROVED** [review]
# 💯 🚢 🚀  🎯 🙏 ❤️

**Review by roseeichelmann — APPROVED** [review]
bet our users will be super happy to have this improvement :)) :rocket:

**Comment by tillyw** [comment]
> Looks and works great! Could you add the same behavior to `EditAttributesModal`?

added! thanks Mike!

**Review by mddilley — APPROVED** [review]
Works great - thanks Tilly! 🚀 🚢

**Pull request merged**